.github/workflows/ci.yml

```diff
@@ -133,13 +152,36 @@ jobs:
           ./build.sh
           # free up space used by libs; our desired clang binary statically links them
           rm -rf build/lib/*.a
+        # if clang was cached, move it into the git tree to the same location as it would be built
+      - name: Move cached Clang
+        if: steps.cache-clang.outputs.cache-hit == 'true' && steps.cache-rtlibs.outputs.cache-hit != 'true'
+        working-directory: ${{ github.workspace }}
+        run: |
+          set -x
+          pwd
+          mkdir -p llvm-project
+          mv llvm-project-built/* llvm-project/
+      - name: Build runtime libraries
+        if: steps.cache-rtlibs.outputs.cache-hit != 'true'
+        working-directory: ${{ github.workspace }}/llvm-project
+        run: |
+          ./cross-build-rtlibs.sh
+          mv build-rtlibs ../
+        # move rtlibs into the git tree to where IA2 build expects them
+      - name: Move cached runtime libraries
+        working-directory: ${{ github.workspace }}/llvm-project
+        run: |
+          mv ../build-rtlibs ./
       - name: Test ARM build
         run: |
           LLVM_DIR=`llvm-config-15 --cmakedir`
           Clang_DIR=`realpath $LLVM_DIR/../clang`
           mkdir build
           pushd build
-          cross_flags="-isystem /usr/aarch64-linux-gnu/include --gcc-toolchain=/usr -march=armv8+memtag"
+          rtlibs_dir="$(pwd)/../llvm-project/build-rtlibs"
+          cross_link_flags="-B${rtlibs_dir}/compiler-rt/lib/linux -L${rtlibs_dir}/lib \
+          --rtlib=compiler-rt --unwindlib=libunwind"
+          cross_flags="-isystem /usr/aarch64-linux-gnu/include --gcc-toolchain=/usr -march=armv8.5-a+memtag -ffixed-x18 --rtlib=compiler-rt --stdlib=libc++ -I${rtlibs_dir}/include/c++/v1"
```

Comment:
Either break both sets of flags or have long lines in both places (I think long lines are fine here).

Reply:
I'll break them for readability.